### PR TITLE
executor: remove Next() function for CheckTableExec

### PR DESCRIPTION
**Pull request by qxhy123**

### CODE REVIEW

File: executor/executor.go [hunk 1]
```diff
@@ -363,16 +363,6 @@ func (e *CheckTableExec) Open(ctx context.Context) error {
 	return nil
 }
 
-// Next implements the Executor Next interface.
-func (e *CheckTableExec) Next(ctx context.Context) (Row, error) {
-	if e.done {
-		return nil, nil
-	}
-	err := e.run(ctx)
```
Comment (zz-jason) [inline executor/executor.go]: we can remove `e.run` and put its code inside `NextChunk`

Comment (qxhy123) [reply executor/executor.go]: ok

File: executor/executor.go [hunk 2]
```diff
@@ -398,6 +382,7 @@ func (e *CheckTableExec) run(ctx context.Context) error {
 			}
 		}
 	}
+	e.done = true
```
Comment (shenli) [inline executor/executor.go]: This is different from the current behaviour. You can use defer to set e.done to true after line 370. So e.done will be set to true whether meet error or not.

**Comment by zz-jason** [comment]
@shenli PTAL

**Comment by shenli** [comment]
LGTM

**Pull request merged**